basic/airdrop-with-withdraw/tests/lib.rs

```diff
@@ -0,0 +1,365 @@
+    
+    let token_by_recipient =  Decimal::from(10); 
+    let (_, recipient_account)  = test_env.new_account();
+    
+    let first_add_receipt = test_env.add_recipient(recipient_account, token_by_recipient); 
+    assert!(first_add_receipt.success);
+
+    let second_add_receipt = test_env.add_recipient(recipient_account, token_by_recipient); 
+    let log_message = &second_add_receipt.logs.get(0).unwrap().1;
+    assert!(!second_add_receipt.success);
+    assert!(log_message.starts_with("Panicked at 'Recipient already exist'"));
+}
+
+#[test]
+fn try_put_airdrop_tokens_without_recipient_must_be_failed() {
+
+    let mut ledger = InMemoryLedger::with_bootstrap();
+    // Set up environment.
+    let mut test_env = TestEnv::new(&mut ledger, RADIX_TOKEN);
+    assert_eq!(test_env.get_balance(test_env.admin_account, RADIX_TOKEN).unwrap(), Decimal::from_str("1000000").unwrap());
+    
+    let airdrop_tokens = Decimal::from(1000); 
+    let receipt = test_env.put_airdrop_tokens(airdrop_tokens, RADIX_TOKEN);
+    assert!(!receipt.success);
+    
+    let log_message = &receipt.logs.get(0).unwrap().1;
+    assert!(log_message.starts_with("Panicked at 'You must register at least one recipient before put tokens'"));
+    assert_eq!(test_env.get_balance(test_env.admin_account, RADIX_TOKEN).unwrap(), Decimal::from_str("1000000").unwrap());
+}
+
+#[test]
+fn try_put_airdrop_tokens_with_insufficient_tokens_must_be_failed() {
+    // Set up environment.
+    let mut ledger = InMemoryLedger::with_bootstrap();
+    // Set up environment.
+    let token_by_recipient =  Decimal::from(10); 
+    let mut test_env = TestEnv::new(&mut ledger, RADIX_TOKEN);
+    assert_eq!(test_env.get_balance(test_env.admin_account, RADIX_TOKEN).unwrap(), Decimal::from_str("1000000").unwrap());
+
+    let recipient_count = 3; 
+    for _ in 0..(recipient_count) {
+        let (_, recipient_account)  = test_env.new_account();
+        test_env.add_recipient(recipient_account, token_by_recipient); 
+    }
+
+    let airdrop_tokens = Decimal::from(20);
+    let receipt = test_env.put_airdrop_tokens(airdrop_tokens, RADIX_TOKEN);
+    assert!(!receipt.success);
+    
+    let log_message = &receipt.logs.get(0).unwrap().1;
+    assert!(log_message.starts_with("Panicked at 'The tokens quantity must be 30'"),);
+    assert_eq!(test_env.get_balance(test_env.admin_account, RADIX_TOKEN).unwrap(), Decimal::from_str("1000000").unwrap());
+}
+
+
+#[test]
+fn try_put_airdrop_tokens_with_sufficient_tokens_must_be_succeeded() {
+    // Set up environment.
+    let mut ledger = InMemoryLedger::with_bootstrap();
+    // Set up environment.
+    let token_by_recipient : Decimal =  Decimal::from(100); 
+    let mut test_env = TestEnv::new(&mut ledger, RADIX_TOKEN);
+    assert_eq!(test_env.get_balance(test_env.admin_account, RADIX_TOKEN).unwrap(), Decimal::from_str("1000000").unwrap());
```

Comment:
Instead of `Decimal::from_str().unwrap`, you can use `Decimal::from(1000000)` instead

Reply:
done